stream-chat-android-client/src/main/java/io/getstream/chat/android/client/controllers/ChannelController.kt

```diff
@@ -152,22 +187,32 @@ public interface ChannelController {
     public fun subscribeFor(
         lifecycleOwner: LifecycleOwner,
         vararg eventTypes: Class<out ChatEvent>,
-        listener: (event: ChatEvent) -> Unit
+        listener: ChatEventListener<ChatEvent>,
+    ): Disposable
+
+    @Deprecated(
+        message = "Use subscribeFor with ChatEventListener parameter",
+        level = DeprecationLevel.WARNING
+    )
+    public fun subscribeFor(
+        lifecycleOwner: LifecycleOwner,
+        vararg eventTypes: Class<out ChatEvent>,
+        listener: (ChatEvent) -> Unit,
     ): Disposable
 
     /**
      * Subscribes for the next channel event with the given [eventType].
      *
      * @see [io.getstream.chat.android.client.models.EventType] for type constants
      */
-    public fun subscribeForSingle(eventType: String, listener: (event: ChatEvent) -> Unit): Disposable
+    public fun subscribeForSingle(eventType: String, listener: ChatEventListener<ChatEvent>): Disposable
```

Comment:
There are some method here that don't have the old declarations anymore that take a function as a parameter, just the new one